Code/Framework/AzToolsFramework/AzToolsFramework/Entity/PrefabEditorEntityOwnershipService.cpp

```diff
@@ -104,39 +105,62 @@ namespace AzToolsFramework
     void PrefabEditorEntityOwnershipService::AddEntity(AZ::Entity* entity)
     {
         AZ_Assert(IsInitialized(), "Tried to add an entity without initializing the Entity Ownership Service");
-        ScopedUndoBatch undoBatch("Undo adding entity");
+
+        // Determine which prefab instance should own this entity
+        AZ::EntityId currentParentId;
+        AZ::TransformBus::EventResult(currentParentId, entity->GetId(), &AZ::TransformInterface::GetParentId);
+
+        Prefab::InstanceOptionalReference newOwningInstance;
+        // If the entity has a parent set, try to find that entity's owning instance and use that.
+        if (currentParentId.IsValid())
+        {
+            auto* instanceEntityMapperInterface = AZ::Interface<Prefab::InstanceEntityMapperInterface>::Get();
+            if (instanceEntityMapperInterface)
```

Comment:
suggestion: Could either reduce scope or it might be worth asserting `instanceEntityMapperInterface` is not null as something is horribly wrong if that is the case I imaging (instead of silently failing).